src/contracts/account/ProxyAccountDeployer.sol

```diff
@@ -22,63 +36,83 @@ contract ProxyAccount is ReplayProtection {
     }
 
     /**
-     * Each signer has a proxy account (signers address => contract address).
-     * We check the signer has authorised the target contract and function call. Then, we pass it to the
-     * signer's proxy account to perform the final execution (to help us bypass msg.sender problem).
-     * @param _target Target contract
-     * @param _value Quantity of eth in account contract to send to target
-     * @param _callData Function name plus arguments
-     * @param _replayProtection Replay protection (e.g. multinonce)
-     * @param _replayProtectionAuthority Identify the Replay protection, default is address(0)
+     * We check the signature has authorised the call before executing it.
+     * @param _metaTx A single meta-transaction that includes target, value and calldata
+     * @param _replayProtection Replay protection
+     * @param _replayProtectionAuthority Address of external replay protection
      * @param _signature Signature from signer
      */
     function forward(
-        address _target,
-        uint _value, 
-        bytes memory _callData,
+        MetaTx memory _metaTx,
         bytes memory _replayProtection,
         address _replayProtectionAuthority,
         bytes memory _signature) public {
 
         // Assumes that ProxyAccountDeployer is ReplayProtection. 
-        bytes memory encodedData = abi.encode(_target, _value, _callData);
+        bytes memory encodedData = abi.encode(CallType.CALL, _metaTx.target, _metaTx.value, _metaTx.callData);
 
         // // Reverts if fails.
         require(owner == verify(encodedData, _replayProtection, _replayProtectionAuthority, _signature), "Owner did not sign this meta-transaction.");
+        forwardCall(_metaTx.target, _metaTx.value, _metaTx.callData); // We do not want this to revert. Save the replay protection 
```

Comment:
ya can return success

Reply:
i made it return success for forward and delegate. it doesnt make too much sense for batch, e.g. since it can potentially revert. the .call() into batch should return true or false anyway.

Reply:
I meant return the actual payload of the call. Like a call, return `(bool success, bytes result)`

Reply:
it now returns result of payload. 

the batch() doesnt. dno if it makes sense to return a list of all returns? esp if we have the revert option